singularity-eos/eos/eos_vinet.hpp

```diff
@@ -0,0 +1,374 @@
+  Real output[8];
+  Vinet_F_DT_func(rho, temp, output);
+  return output[0];
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::PressureFromDensityTemperature(const Real rho,
+                                                                    const Real temp,
+                                                                    Real *lambda) const {
+  Real output[8];
+  Vinet_F_DT_func(rho, temp, output);
+  return output[1];
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::EntropyFromDensityTemperature(const Real rho,
+                                                                   const Real temp,
+                                                                   Real *lambda) const {
+  Real output[8];
+  Vinet_F_DT_func(rho, temp, output);
+  return output[6];
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::TExpansionCoeffFromDensityTemperature(
+    const Real rho, const Real temp, Real *lambda) const {
+  Real output[8];
+  Vinet_F_DT_func(rho, temp, output);
+  return output[3] / output[2] / rho;
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::TBulkModulusFromDensityTemperature(
+    const Real rho, const Real temp, Real *lambda) const {
+  Real output[8];
+  Vinet_F_DT_func(rho, temp, output);
+  return output[2] * rho;
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::BulkModulusFromDensityTemperature(
+    const Real rho, const Real temp, Real *lambda) const {
+  Real output[8];
+  Vinet_F_DT_func(rho, temp, output);
+  return output[7] * output[7] * rho;
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::TemperatureFromDensityInternalEnergy(
+    const Real rho, const Real sie, Real *lambda) const {
+  Real Tref;
+  Real output[8];
+  Tref = _T0;
+  Vinet_F_DT_func(rho, Tref, output);
+  return (sie - output[0]) / _Cv0 + Tref;
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::PressureFromDensityInternalEnergy(
+    const Real rho, const Real sie, Real *lambda) const {
+  Real temp;
+  Real output[8];
+  temp = TemperatureFromDensityInternalEnergy(rho, sie);
+  Vinet_F_DT_func(rho, temp, output);
+  return output[1];
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::EntropyFromDensityInternalEnergy(
+    const Real rho, const Real sie, Real *lambda) const {
+  Real temp;
+  Real output[8];
+  temp = TemperatureFromDensityInternalEnergy(rho, sie);
+  Vinet_F_DT_func(rho, temp, output);
+  return output[6];
+}
+PORTABLE_INLINE_FUNCTION Real Vinet::BulkModulusFromDensityInternalEnergy(
+    const Real rho, const Real sie, Real *lambda) const {
+  Real temp;
+  Real output[8];
+  temp = TemperatureFromDensityInternalEnergy(rho, sie);
+  Vinet_F_DT_func(rho, temp, output);
+  return output[7] * output[7] * rho;
+}
+// AEM: Give error since function is not well defined
+PORTABLE_INLINE_FUNCTION void
+Vinet::DensityEnergyFromPressureTemperature(const Real press, const Real temp,
+                                            Real *lambda, Real &rho, Real &sie) const {
+  EOS_ERROR("Vinet::DensityEnergyFromPressureTemperature: "
+            "Not implemented.\n");
+}
```

Comment:
This is fine we should maybe delete this function.

Reply:
I think we can just cover this under https://github.com/lanl/singularity-eos/issues/199